src/hooks/react/useDebounceState.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import { useCallback, useState } from 'react';
+import debounce from 'lodash/debounce';
```

Comment:
just `import { debounce } from 'lodash';` will do the job

Reply:
I think we discussed it before , but doing that you are importing the whole lodash package which is quite big.

Reply:
@hbulgarini @wirednkod please compare numbers not opinions. I agree with Nikos here, simple import is nicer and I find it hard to believe that the code cannot be tree-shaken efficiently. However if you can prove to me that the artifacts are *significantly* larger if we do that I'm happy to rewrite pretty much everything to the approach you propose.

Reply:
@hbulgarini thanks! I was skeptical because the article is quite old (May 2018), and run tests myself. We only have 3 instances of `lodash` usage currently and all of them is `isEmpty`. One in `src/App.tsx` is not using the syntax you've mentioned - fixing this (as done in https://github.com/paritytech/parity-bridges-ui/pull/214) reduces the build artifacts by 23KBs. I stand corrected.

Reply:
cool!

Reply:
I know that @tomusdrw will call me "too opinionated" but _just For Our Information (including some good will)_, this is a characteristic of lodash that - afaiu - [will not longer be supported on v5 as per official docs](https://lodash.com/per-method-packages). In addition - as I do not like CRA so much and prefer other bundlers - webpack and parcel transpiles the above mentioned tree shaking absolutely perfectly while CRA does not

😃